### PR TITLE
OOP SCHOOL LIBRARY: Associations setup

### PR DESCRIPTION
**Pull request by ngalamac**
In this project, the following were implemented:
-1. Created the following classes in separate files
- Classroom
- Book
- Rental
2. Established the appropriate relationships between classes as required

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
```
Comment (MahmoudBakr23) [inline rental.rb]: - [ ] Great job in the Rental's associations and the initializer method. However, you missed adding the lines of code that dictate the Rental class to add itself to the book and person provoked. 🙏 

EX:
`book.rentals << self`
_the same for person_

File: student.rb [hunk 2]
```diff
@@ -2,9 +2,12 @@
 require_relative 'person'
 
 class Student < Person
```
Comment (MahmoudBakr23) [inline student.rb:4]: - [ ] I see you've not updated your Person class so I'll leave the suggestion here. Please make sure to add the necessary associations between the Person class and the Rental class as you did in the Book class. 🙏 
- [ ] Please make sure to create the `add_rental` method in the Person class properly. 👍

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @ngalamac 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done creating the associations between the classes so far
- Good job with the custom methods
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 3]
```diff
@@ -31,4 +32,8 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (mikethreels) [inline person.rb]: - [ ] good job adding the `add_rental` method, however, the method should create a new instance of Rental, have a look at [the example in a previous lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship). you can achieve this by doing the following:
```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end

```

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (mikethreels) [inline book.rb]: - [ ] good job creating the `add_rental` method, however, the method should create a new instance of Rental, have a look at [the example in a previous lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship). you can achieve this by doing the following:
```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end

```

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi @ngalamac, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Good job implementing some of the requested changes

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @ngalamac , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the change requested to ensure the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental is correctly implemented.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**